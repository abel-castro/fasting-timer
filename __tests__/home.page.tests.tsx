import { render, screen, fireEvent, act } from "@testing-library/react";
import Home from "../app/page";
import React from "react";
import storageService from "../app/storage/storage-service";

describe("Page", () => {
  it("renders initial state unchanged", () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });

  it("should start the timer and switch to fasting/eating mode after clicking the button", async () => {
    const { container } = render(<Home />);

    expect(container).toMatchSnapshot();

    // Start fasting
    const button = screen.getByRole("button", { name: /Start fasting/i });
    await act(async () => {
      fireEvent.click(button);
    });

    expect(container).toMatchSnapshot();

    const startFastingTime = storageService.getStartFastingTime();
    expect(startFastingTime).toBeInstanceOf(Promise<Date>);

    // Stop fasting / start eating
    await act(async () => {
      fireEvent.click(button);
    });

    expect(container).toMatchSnapshot();

    const startEatingTime = storageService.getStartEatingTime();
    expect(startEatingTime).toBeInstanceOf(Promise<Date>);

    // Reset timer
    const resetButton = screen.getByRole("button", { name: /Reset timer/i });
    await act(async () => {
      fireEvent.click(resetButton);
    });

    expect(container).toMatchSnapshot();

    expect(storageService.getStartFastingTime()).toBeInstanceOf(Promise<{}>);
    expect(storageService.getStartEatingTime()).toBeInstanceOf(Promise<{}>);
  });
});
