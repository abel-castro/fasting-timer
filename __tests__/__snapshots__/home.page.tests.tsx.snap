// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders initial state unchanged 1`] = `
<div>
  <main
    class="min-h-screen flex items-center justify-center"
  >
    <div
      class="p-4 max-w-xl w-full mx-auto"
    >
      <header
        class="p-4 mb-16"
      >
        <h1
          class="text-3xl text-center"
        >
          Fasting Timer 18:6
        </h1>
      </header>
      <div
        class="flex flex-col items-center justify-center"
      >
        <div
          class="w-full"
        >
          <p
            class="flex flex-col items-center justify-center mt-16"
          >
            Press start for initiate the timer
          </p>
        </div>
        <form
          class="p-4"
        >
          <button
            class="text-white font-bold py-2 px-4 border-b-4 border-blue-700 hover:border-blue-500 rounded bg-blue-500 hover:bg-blue-700 hover:bg-blue-700 hover:border-blue-500"
            type="submit"
          >
            Start fasting
          </button>
        </form>
      </div>
      <footer
        class="text-slate-300 mt-48 text-center text-sm"
      >
        <p>
          Abel Castro 
          2024
           - checkout the source code on
           
          <a
            class="footer-link text-teal-500 hover:text-teal-700"
            href="https://github.com/abel-castro/fasting-timer"
            target="_blank"
          >
            GitHub
          </a>
           
        </p>
      </footer>
    </div>
  </main>
</div>
`;

exports[`Page should start the timer and switch to fasting/eating mode after clicking the button 1`] = `
<div>
  <main
    class="min-h-screen flex items-center justify-center"
  >
    <div
      class="p-4 max-w-xl w-full mx-auto"
    >
      <header
        class="p-4 mb-16"
      >
        <h1
          class="text-3xl text-center"
        >
          Fasting Timer 18:6
        </h1>
      </header>
      <div
        class="flex flex-col items-center justify-center"
      >
        <div
          class="w-full"
        >
          <p
            class="flex flex-col items-center justify-center mt-16"
          >
            Press start for initiate the timer
          </p>
        </div>
        <form
          class="p-4"
        >
          <button
            class="text-white font-bold py-2 px-4 border-b-4 border-blue-700 hover:border-blue-500 rounded bg-blue-500 hover:bg-blue-700 hover:bg-blue-700 hover:border-blue-500"
            type="submit"
          >
            Start fasting
          </button>
        </form>
      </div>
      <footer
        class="text-slate-300 mt-48 text-center text-sm"
      >
        <p>
          Abel Castro 
          2024
           - checkout the source code on
           
          <a
            class="footer-link text-teal-500 hover:text-teal-700"
            href="https://github.com/abel-castro/fasting-timer"
            target="_blank"
          >
            GitHub
          </a>
           
        </p>
      </footer>
    </div>
  </main>
</div>
`;

exports[`Page should start the timer and switch to fasting/eating mode after clicking the button 2`] = `
<div>
  <main
    class="min-h-screen flex items-center justify-center"
  >
    <div
      class="p-4 max-w-xl w-full mx-auto"
    >
      <header
        class="p-4 mb-16"
      >
        <h1
          class="text-3xl text-center"
        >
          Fasting Timer 18:6
        </h1>
      </header>
      <div
        class="flex flex-col items-center justify-center"
      >
        <div
          class="w-full"
        >
          <div
            class="mb-1 font-medium text-gray-300 text-sm text-center"
          >
            0.00%
             from 
            18
             hours
          </div>
          <div
            class="w-full bg-gray-200 rounded-full h-2.5 mb-4 dark:bg-gray-700"
          >
            <div
              class="bg-green-600 h-2.5 rounded-full dark:bg-green-500"
              style="width: 0.00%;"
            />
          </div>
          <div
            class="mb-1 font-medium text-gray-300 text-sm text-center"
          >
            0%
             from 
            6
             hours
          </div>
          <div
            class="w-full bg-gray-200 rounded-full h-2.5 mb-4 dark:bg-gray-700"
          >
            <div
              class="bg-gray-600 h-2.5 rounded-full dark:bg-gray-300"
              style="width: 0%;"
            />
          </div>
          <div
            class="flex flex-col items-center justify-center mt-16"
          >
            <p>
              In fasting period since:
            </p>
            <p>
              0
               hours
            </p>
            <p>
              0
               minutes
            </p>
            <p>
              0
               seconds
            </p>
          </div>
        </div>
        <form
          class="p-4"
        >
          <button
            class="text-white font-bold py-2 px-4 border-b-4 border-blue-700 hover:border-blue-500 rounded bg-red-500 border-red-700 hover:bg-red-700 hover:border-red-500"
            type="submit"
          >
            Stop fasting
          </button>
        </form>
      </div>
      <footer
        class="text-slate-300 mt-48 text-center text-sm"
      >
        <p>
          Abel Castro 
          2024
           - checkout the source code on
           
          <a
            class="footer-link text-teal-500 hover:text-teal-700"
            href="https://github.com/abel-castro/fasting-timer"
            target="_blank"
          >
            GitHub
          </a>
           
        </p>
      </footer>
    </div>
  </main>
</div>
`;

exports[`Page should start the timer and switch to fasting/eating mode after clicking the button 3`] = `
<div>
  <main
    class="min-h-screen flex items-center justify-center"
  >
    <div
      class="p-4 max-w-xl w-full mx-auto"
    >
      <header
        class="p-4 mb-16"
      >
        <h1
          class="text-3xl text-center"
        >
          Fasting Timer 18:6
        </h1>
      </header>
      <div
        class="flex flex-col items-center justify-center"
      >
        <div
          class="w-full"
        >
          <div
            class="mb-1 font-medium text-gray-300 text-sm text-center"
          >
            0.00%
             from 
            18
             hours
          </div>
          <div
            class="w-full bg-gray-200 rounded-full h-2.5 mb-4 dark:bg-gray-700"
          >
            <div
              class="bg-green-600 h-2.5 rounded-full dark:bg-green-500"
              style="width: 0.00%;"
            />
          </div>
          <div
            class="mb-1 font-medium text-gray-300 text-sm text-center"
          >
            0.00%
             from 
            6
             hours
          </div>
          <div
            class="w-full bg-gray-200 rounded-full h-2.5 mb-4 dark:bg-gray-700"
          >
            <div
              class="bg-gray-600 h-2.5 rounded-full dark:bg-gray-300"
              style="width: 0.00%;"
            />
          </div>
          <div
            class="flex flex-col items-center justify-center mt-16"
          >
            <p>
              In eating period since:
            </p>
            <p>
              0
               hours
            </p>
            <p>
              0
               minutes
            </p>
            <p>
              0
               seconds
            </p>
          </div>
        </div>
        <form
          class="p-4"
        >
          <button
            class="text-white font-bold py-2 px-4 border-b-4 border-gray-700 hover:border-gray-500 rounded bg-gray-500 hover:bg-gray-700 hover:bg-gray-700 hover:border-gray-500"
            type="submit"
          >
            Reset timer
          </button>
        </form>
      </div>
      <footer
        class="text-slate-300 mt-48 text-center text-sm"
      >
        <p>
          Abel Castro 
          2024
           - checkout the source code on
           
          <a
            class="footer-link text-teal-500 hover:text-teal-700"
            href="https://github.com/abel-castro/fasting-timer"
            target="_blank"
          >
            GitHub
          </a>
           
        </p>
      </footer>
    </div>
  </main>
</div>
`;

exports[`Page should start the timer and switch to fasting/eating mode after clicking the button 4`] = `
<div>
  <main
    class="min-h-screen flex items-center justify-center"
  >
    <div
      class="p-4 max-w-xl w-full mx-auto"
    >
      <header
        class="p-4 mb-16"
      >
        <h1
          class="text-3xl text-center"
        >
          Fasting Timer 18:6
        </h1>
      </header>
      <div
        class="flex flex-col items-center justify-center"
      >
        <div
          class="w-full"
        >
          <p
            class="flex flex-col items-center justify-center mt-16"
          >
            Press start for initiate the timer
          </p>
        </div>
        <form
          class="p-4"
        >
          <button
            class="text-white font-bold py-2 px-4 border-b-4 border-blue-700 hover:border-blue-500 rounded bg-blue-500 hover:bg-blue-700 hover:bg-blue-700 hover:border-blue-500"
            type="submit"
          >
            Start fasting
          </button>
        </form>
      </div>
      <footer
        class="text-slate-300 mt-48 text-center text-sm"
      >
        <p>
          Abel Castro 
          2024
           - checkout the source code on
           
          <a
            class="footer-link text-teal-500 hover:text-teal-700"
            href="https://github.com/abel-castro/fasting-timer"
            target="_blank"
          >
            GitHub
          </a>
           
        </p>
      </footer>
    </div>
  </main>
</div>
`;
